VimeoNetworkingExample-iOS/VimeoNetworkingCommonTests/Request+ProgrammedContent.swift

```diff
@@ -0,0 +1,140 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import XCTest
+import OHHTTPStubs
+import VimeoNetworking
+
+class Request_ProgrammedContent: XCTestCase
+{
+    override func setUp()
+    {
+        super.setUp()
+        
+        VimeoClient.configureSharedClient(withAppConfiguration: AppConfiguration(clientIdentifier: "{CLIENT_ID}",
+                                                                          clientSecret: "{CLIENT_SECRET}",
+                                                                          scopes: [.Public, .Private, .Purchased, .Create, .Edit, .Delete, .Interact, .Upload],
+                                                                          keychainService: "com.vimeo.keychain_service",
+                                                                          apiVersion: "3.3.1"))
+    }
+    
+    override func tearDown()
+    {
+        super.tearDown()
+        
+        OHHTTPStubs.removeAllStubs()
+    }
+    
+    func test_CinemaRequest_onSuccess_returnsCorrectData()
+    {
+        let request = CinemaContentRequest.getCinemaContentRequest()
+        
+        stub(condition: isPath("/programmed/cinema")) { _ in
+            // Stub it with our "cinema-success-response.json" stub file (which is in same bundle as self)
+            let stubPath = OHPathForFile("cinema-success-response.json", type(of: self))
+            return fixture(filePath: stubPath!, headers: ["Content-Type":"application/json"])
+        }
+        
+        let expectation = self.expectation(description: "Network call expectation")
+        
+        _ = VimeoClient.sharedClient.request(request) { response in
+            switch response
+            {
+            case .success(let result):
+                XCTAssertNotNil(result.model)
+                
+                let cinemaContent = result.model
+                XCTAssertEqual(cinemaContent.count, 6)
+
+                let cinemaNames = ["John Early's Picks", "Action Sports", "Eye Candy", "Documentary", "Comedy", "The Refugee Crisis"]
+                for i in 0 ..< cinemaNames.count
+                {
+                    XCTAssertEqual(cinemaNames[i], cinemaContent[i].name)
+                    XCTAssertEqual(cinemaContent[i].content?.count, 5)
+                }
+                
+            case .failure(let error):
+                XCTFail("\(error)")
+            }
+            
+            expectation.fulfill()
+        }
+        
+        self.waitForExpectations(timeout: 1.0) { error in
+            if let unWrappedError = error
+            {
+                XCTFail("\(unWrappedError)")
+            }
+        }
+    }
+    
+    func test_CinemaRequest_onFailure_returnsError()
+    {
+        let request = CinemaContentRequest.getCinemaContentRequest()
+        
+        stub(condition: isPath("/programmed/cinema")) { _ in
+            
+            let fakeErrorJSON = ["link": NSNull(),
```

Comment:
This was more of a demo to show a different method for creating the stubbed data.  The info in the JSON file in the previous test was captured from the server using Postman, so it was easy to put into a file.  For this test, I couldn't capture a live failure from the server so this shows how you might enter in error information using a regular Dictionary if you know what keys the code you're testing is looking for.  As for refactoring the file reading code, I'd have to see more examples of it being used before we can see how it should be wrapped.  The OHHttpStubs framework has a convenient method for reading from a file, so it wouldn't make sense to just wrap it with another function unless there was something like validation code that we kept re-writing before using the framework's file reading method.